src/CMakeLists.txt

```diff
@@ -0,0 +1,78 @@
+# ===========================================
+# Set up navier-stokes library
+
+add_library(navier_stokes INTERFACE "navier_stokes/navier_stokes.h")
+add_library(feltor::navier_stokes ALIAS navier_stokes)
+target_link_libraries(navier_stokes INTERFACE dg::dg)
+
+# ============================================
+# Set up FELTOR
+# TODO: Other executables
+
+set(FELTOR_HEADERS
+    "feltor/feltor.h"
+    "feltor/init_from_file.h"
+    "feltor/init.h"
+    "feltor/parameters.h"
+)
+
+add_executable(feltor "feltor/feltor.cpp" ${FELTOR_HEADERS})
+add_executable(feltor::feltor ALIAS feltor)
+
+target_include_directories(feltor PRIVATE ${CMAKE_CURRENT_SOURCE_DIR})
+
+if(FELTOR_USE_MPI)
+    target_link_libraries(feltor PRIVATE MPI::MPI_CXX)
+    target_compile_definitions(feltor PRIVATE WITH_MPI)
+endif()
+
+# TODO: Replace following commands with portable alternatives.
+execute_process(
+    COMMAND git rev-parse HEAD
+    OUTPUT_VARIABLE GIT_HASH
+    OUTPUT_STRIP_TRAILING_WHITESPACE
+    WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}
+)
+message(STATUS "Git hash: ${GIT_HASH}")
+
+execute_process(
+    COMMAND git branch --show-current
+    OUTPUT_VARIABLE GIT_BRANCH
+    OUTPUT_STRIP_TRAILING_WHITESPACE
+    WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}
+)
+message(STATUS "Git branch: ${GIT_BRANCH}")
+
+execute_process(
+    COMMAND "date" "+'%F %T %Z'"
+    OUTPUT_VARIABLE COMPILE_TIME
+    OUTPUT_STRIP_TRAILING_WHITESPACE
+)
+message(STATUS "Compile time: ${COMPILE_TIME}")
+
+target_compile_definitions(feltor PRIVATE 
+    GIT_HASH="${GIT_HASH}"
+    GIT_BRANCH="${GIT_BRANCH}"
+    COMPILE_TIME="${COMPILE_TIME}"
+)
```

Comment:
Did the netcdf tests pass without having these definitions in FeltorTests?

Reply:
It seems I wasn't running the tests in `inc/file`! I'll get that fixed